.github/workflows/rust.yml

```diff
@@ -0,0 +1,54 @@
+on: [push, pull_request]
+
+name: Continuous integration
+
+jobs:
+    tests:
+        name: Tests
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                include:
+                    - rust: stable
+                      env:
+                          RUSTFMTCHK: true
+                    - rust: nightly
+                      env:
+                          RUSTFMTCHK: false
+                    - rust: 1.29.0
+                      env:
+                          PIN_VERSIONS: true
+        steps:
+            - name: Checkout Crate
+              uses: actions/checkout@v2
```

Comment:
Why do we need checkout here? Was always doing the same without checkout

Reply:
My understanding is that this makes the files of the repo available in the current action workspace. I made a try without it and you can see that it can't find the `test.sh` file for example:
https://github.com/p2pderivatives/rust-bitcoincore-rpc/runs/3416665034?check_suite_focus=true

Maybe there is another way to do it?

Reply:
That's my understanding too. Having a step using `actions/checkout@v2` to fetch the repostiry is part of the default GitHub Actions workflow. 

https://github.com/actions/checkout